packages/core/src/tools/create-cache.js

```diff
@@ -20,23 +26,32 @@ const createCache = (input) => {
       throw new TypeError('ttl must be an integer');
     }
 
+    if (
+      scopes != null &&
+      !scopes.every((scope) => scope === 'user' || scope === 'auth')
+    ) {
```

Comment:
If `scopes` is a non-array object, the `TypeError` should be thrown as well. So:

```suggestion
    if (
      !Array.isArray(scopes) ||
      !scopes.every((scope) => scope === 'user' || scope === 'auth')
    ) {
```